packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -487,7 +487,7 @@ const methods = {
     throw new Error('non pool-wide install not implemented')
   },
 
-  async rollingPoolUpdate($defer, { xsCredentials } = {}) {
+  async rollingPoolUpdate($defer, task, { xsCredentials } = {}) {
```

Comment:
I'm not a fan of forwarding the `task` into the function settings.

Since the `progress` of a task is something that will be defined in many cases, what do you think on implementing something to manage the `progress` of the parent task from the child tasks?

/cc @julien-f

Reply:
I don't want to give subtasks the possibility to interact with their parent for now.

I believe the code responsible for creating a task is the one that should be able to interact with it. The difficulty here is that the code is split between task management in the XO mixin and the logic in the XAPI mixin.

I don't have a better solution than what @b-Nollet is doing at this time.